src/client/lazy.rs

```diff
@@ -507,6 +507,7 @@ mod hyper {
     impl<'d> super::PreparedFields<'d> {
         /// #### Feature: `hyper`
         /// Convert `self` to `hyper::client::Body`.
+        #[cfg_attr(feature="clippy", allow(wrong_self_convention))]
```

Comment:
Let it warn on this for now, that way I don't forget to fix this for the next minor version release.